### PR TITLE
Implement getUserFeedbackMap to organize feedback by user and event (TC #2)

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- What does this PR do?  
  Adds a utility function `getUserFeedbackMap` that retrieves all feedback records from the Supabase `event_feedback` table and structures them into a nested object organized by user and event.

- Why is it necessary or valuable?  
  This structured format enables efficient access and filtering for personalized recommendation algorithms. Grouping feedback by user and event makes it easy to analyze individual preferences and aggregate patterns.

- What is intentionally left out and will be done in a later PR?  
  - Vectorization of feedback for clustering  
  - Feedback analytics logic  

---

# Milestones / Related Work  
- Feature or user story this contributes to: [4.3 Develop logic for storing user reactions and feedback reasons  (TC #2)](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.jg0hljkko6h9)

---

# Resources and References  
- Tutorials, documentation, or code examples used:  
  [Supabase Documentation](https://supabase.com/docs)

---

# Test Plan  
- How was this tested?  
  - Logged outputs to verify structure and correctness  

## Edge Cases Tested  
- No feedback entries in database  
- Empty or missing `reasons` array

---

# Additional Notes  
- Planned follow-up PRs:  
  - Convert structured map into vector format for clustering  
  - Integrate with recommendation alg

### CODE REVIEW

File: frontend/src/utils/feedbackUtils.js [hunk 1]
```diff
@@ -11,3 +12,26 @@ export async function saveUserFeedback(userId, eventId, liked, reasonsArray) {
...
+			console.error("error fetching feedback:", error.message);
+			return {};
+		}
+		const map = {};
+		data.forEach(({ user_id, event_id, liked, reasons }) => {
+			const uid = user_id;
+			const eid = String(event_id);
+			if (!map[uid]) map[uid] = {};
+      // store liked status and convert reasons array to a set for fast lookup
+			map[uid][eid] = { liked, reasons: new Set(reasons) };
```
Comment (weibinma123) [inline frontend/src/utils/feedbackUtils.js:30]: similar to my comment in the previous commit, is there any reason why we don't include `liked` into the key?
